src/ClearArchivedRecords.php

```diff
@@ -121,8 +129,10 @@ public function flushClass(string $class)
     {
         $this->message("Beginning flush for {$class}");
 
-        // Delete old versions for non-deleted records
-        $this->deleteOldVersions($class);
+        // Delete old versions for non-deleted records (note: Can be slow on large recordsets)
+        if (!$this->isFast()) {
```

Comment:
Tempted to say you should use `isSlow` logic here to simplify but that's all the feedback I have :)